contracts/internal/EverlongERC4626.sol

```diff
@@ -54,6 +34,97 @@ abstract contract EverlongERC4626 is ERC4626 {
         _decimals = success ? result : _DEFAULT_UNDERLYING_DECIMALS;
     }
 
+    // ╭─────────────────────────────────────────────────────────╮
+    // │ Stateful                                                │
+    // ╰─────────────────────────────────────────────────────────╯
+
+    /// @inheritdoc ERC4626
+    function deposit(
+        uint256 assets,
+        address to
+    ) public override returns (uint256 shares) {
+        IERC20(_asset).approve(hyperdrive, assets);
```

Comment:
The best practice is generally to avoid dangling approvals. I get that we are going to do the rebalance anyways after deposit, but why don't we do the approval there before we use it? IMO that would make the code more readable.

Reply:
If we eliminate this, I think we can delete this function.

Reply:
You're totally right, don't need this at all since there's already an approval in rebalance.